### PR TITLE
Updated to intel18 on toss3

### PR DESCRIPTION
**Pull request by bathmatt**
Whoops, needed to set to develop, not master,

@bartlettroscoe any issues?

### CODE REVIEW

**Comment by bathmatt** [comment]
OK, odd, I tried it on serrano..  Ugh, i hate our computers..

**Comment by bathmatt** [comment]
This is an issue with the sems modules on chama and most likely other systems.

You can load it with a clean env

```
    module load sems-devpack-intel/18.0.0 
    module swap mkl/18.0.0.128 mkl/18.0.5.274
    module load gcc/4.9.3
    module unload sems-python/2.7.9

```

but then try to reload purge has those warnings

```

[mbetten@chama-login3 Trilinos]$ module purge
        ERROR loading module sems-boost/1.63.0/base
        Module points to path  /projects/sems/install/toss3/sems/tpl/boost/1.63.0/intel/LOCAL_VERSION_NOT_AVAILABLE/base  which does not exist
        You may need to load the appropriate compilers before loading this module
Lmod Warning:  /projects/sems/modulefiles/toss3/sems/tpl/sems-boost/1.63.0/base: (sems-boost/1.63.0/base): invoked "break" outside of a loop 
While processing the following module(s):
    Module fullname         Module Filename
    ---------------         ---------------
    sems-boost/1.63.0/base  /projects/sems/modulefiles/toss3/sems/tpl/sems-boost/1.63.0/base


        ERROR loading module sems-hdf5/1.8.12/parallel
        Module points to path  /projects/sems/install/toss3/sems/tpl/hdf5/1.8.12/intel/LOCAL_VERSION_NOT_AVAILABLE/openmpi/LOCAL_VERSION_NOT_AVAILABLE/parallel  which does not exist
        You may need to load the appropriate compilers before loading this module
Lmod Warning:  /projects/sems/modulefiles/toss3/sems/tpl/sems-hdf5/1.8.12/parallel: (sems-hdf5/1.8.12/parallel): invoked "break" outside of a loop 
While processing the following module(s):
    Module fullname            Module Filename
    ---------------            ---------------
    sems-hdf5/1.8.12/parallel  /projects/sems/modulefiles/toss3/sems/tpl/sems-hdf5/1.8.12/parallel



```

**Comment by bartlettroscoe** [comment]
@bathmatt said:

> but then try to reload purge has those warnings

Does this mean that we need to remove the `module purge` command like we do for 'mutrino'?

**Comment by bathmatt** [comment]
I submitted a SEMS bug report, I think this can go forward since it isn't an issue with this commit but with SEMS modules.

**Comment by bathmatt** [comment]
SEMSHELPD-1409 sems modules on chama

**Comment by bathmatt** [comment]
No, sems purge still purges, it just prints a warning.  Let me look at something really quick..

**Comment by bathmatt** [comment]
I am pushing a fix for the output issues.

**Comment by bathmatt** [comment]
I think this is ready to merge now... @bartlettroscoe

**Comment by bartlettroscoe** [comment]
@bathmatt said:

> I think this is ready to merge now... @bartlettroscoe

Let me do a little testing on 'chama' and 'serrano' and if that looks good, we can merge.

**Comment by bartlettroscoe** [comment]
@bathmatt ,

FYI:

I am still seeing errors like:

````
        ERROR loading module sems-boost/1.66.0/base
        Module points to path  /projects/sems/install/toss3/sems/tpl/boost/1.66.0/intel/LOCAL_VERSION_NOT_AVAILABLE/base  which does not exist
        You may need to load the appropriate compilers before loading this module
Lmod Warning:
/projects/sems/modulefiles/toss3/sems/tpl/sems-boost/1.66.0/base:
(sems-boost/1.66.0/base): invoked "break" outside of a loop 
While processing the following module(s):
    Module fullname         Module Filename
    ---------------         ---------------
    sems-boost/1.66.0/base  /projects/sems/modulefiles/toss3/sems/tpl/sems-boost/1.66.0/base


        ERROR loading module sems-hdf5/1.8.12/parallel
        Module points to path  /projects/sems/install/toss3/sems/tpl/hdf5/1.8.12/intel/LOCAL_VERSION_NOT_AVAILABLE/openmpi/LOCAL_VERSION_NOT_AVAILABLE/parallel  which does not exist
        You may need to load the appropriate compilers before loading this module
Lmod Warning:
/projects/sems/modulefiles/toss3/sems/tpl/sems-hdf5/1.8.12/parallel:
(sems-hdf5/1.8.12/parallel): invoked "break" outside of a loop 
While processing the following module(s):
    Module fullname            Module Filename
    ---------------            ---------------
    sems-hdf5/1.8.12/parallel  /projects/sems/modulefiles/toss3/sems/tpl/sems-hdf5/1.8.12/parallel
```

but if you and your users are okay with this, if the testing I am running passes, we can go ahead and merge.

File: cmake/std/atdm/common/toss3/environment.sh [hunk 1]
```diff
@@ -18,6 +18,9 @@ export ATDM_CONFIG_BUILD_COUNT=8
 # NOTE: Above, currently setting CMAKE_JOB_POOL_LINK results in a build
 # failures with Ninja.  See https://gitlab.kitware.com/snl/project-1/issues/60
 
+# We do this twice since sems modules are wacked and we get errors to the screen on a purge
+# The second purge will catch any real errors with purging ...
+module purge &> /dev/null
 module purge
```
Comment (jmgate) [inline cmake/std/atdm/common/toss3/environment.sh]: Perhaps you want to remove this line and change the comment above accordingly.

Comment (bathmatt) [reply cmake/std/atdm/common/toss3/environment.sh]: I added that line and made the comment, what don't you like about that line?

Comment (bathmatt) [reply cmake/std/atdm/common/toss3/environment.sh]: @bartlettroscoe can you send me exactly what you are doing to get those warnings?

Comment (jmgate) [reply cmake/std/atdm/common/toss3/environment.sh]: Nevermind @bathmatt&mdash;I understand what you were going for now.

**Comment by bartlettroscoe** [comment]
@bathmatt asked:

> @bartlettroscoe can you send me exactly what you are doing to get those warnings?

I can't explain it, but I don't see those warnings on 'chama' or 'serrano' when I just tried just now.

I am enabling a few key Trilinos packages (Kokkos, Teuchos, SEACAS and Panzer) and testing on 'chama' and 'serrano' posting to CDash shown [here](https://testing.sandia.gov/cdash-dev-view/index.php?project=Trilinos&date=2019-03-20&filtercount=2&showfilters=1&filtercombine=and&field1=buildname&compare1=65&value1=Trilinos-atdm-&field2=groupname&compare2=61&value2=Experimental).

Is this urgent to merge to the Trilinos 'develop' branch right now or can we let these builds run and give back some test results?  If these pass (or not have any major errors) I can make sure this gets merged today.  Okay?

**Comment by bathmatt** [comment]
no idea on why you'd see that, the first purge goes to a black hole.  

Serrano doesn't work until this is patched so . . .

**Comment by bartlettroscoe** [comment]
@bathmatt said:

> Serrano doesn't work until this is patched so . . .

The file cmake/std/atdm/common/toss3/environment.sh is used for both the 'chama' and the 'serrano' envs.  See:

* https://github.com/trilinos/Trilinos/blob/develop/cmake/std/atdm/serrano/environment.sh

Therefore, when one changes cmake/std/atdm/common/toss3/environment.sh one is changing both 'chama' and 'serrano' (soon to be called the 'tlcc2' and 'cts1' envs).

**Comment by bathmatt** [comment]
@bartlettroscoe I have no idea what I need to do to fix this.  it seems to work fine for me on chama and serrano...  we can't build empire on either of these until this patch goes thorugh.  I don't know how to reproduce your issue.  tell me the problem and I'll fix it..

**Comment by bartlettroscoe** [comment]
@bathmatt said:

> I have no idea what I need to do to fix this.

As I mention [above](https://github.com/trilinos/Trilinos/pull/4666#issuecomment-475001348), there is nothing known wrong at point.  I am just running some tests to make sure that there is nothing major wrong with Trilinos with this upgrade.  (Upgrades of compilers on other machines have resulted in problems.)

If EMIRE needs this urgently, we can rebase this branch on top of `em-plasma/trilinos-develop` and directly merge it to `em-plasma/develop` so EMPIRE can start using this ASAP.  Let me and I can do that right now.  Then once the testing on the Trilinos side is completed and there are no major problems, we can merge this branch.

Let me know

**Comment by bartlettroscoe** [comment]
I tested this on 'chama' with:

```
$ cd /nscratch/rabartl/Trilinos.base/BUILDS/CHAMA/CTEST_S/

$ hostname
chama-login8

$ screen -S ctest-s

$ env Trilinos_PACKAGES=Kokkos,Teuchos,SEACAS,Panzer \
  ./ctest-s-local-test-driver.sh all \
  $> ctest-s-local-test-driver.out &
```

which submitted 100% passing to CDash here:

* [Trilinos-atdm-chama-intel-debug-openmp-exp](https://testing.sandia.gov/cdash-dev-view/index.php?project=Trilinos&parentid=4758867)

and I tested on 'serrano' with:

```
$ cd /nscratch/rabartl/Trilinos.base/BUILDS/SERRANO/CTEST_S/

$ hostname
serrano-login6

$ screen -S ctest-s

$ env Trilinos_PACKAGES=Kokkos,Teuchos,SEACAS,Panzer \
  ./ctest-s-local-test-driver.sh all \
  $> ctest-s-local-test-driver.out &
```

which submitted 100% passing to CDash here:

* [Trilinos-atdm-serrano-intel-debug-openmp-exp](https://testing.sandia.gov/cdash-dev-view/index.php?project=Trilinos&parentid=4758867)

That did not test all of that ATDM Trilinos packages but given that we have other Intel 18 builds on other platforms I think this is pretty safe to merge this based on these results.

**Pull request merged**